containers/src/Data/IntMap/Internal.hs

```diff
@@ -2501,7 +2501,7 @@ maxView m = let (k, a) = findMax m
 -- > minViewWithKey empty == Nothing
 minViewWithKey :: IntMap a -> Maybe ((Key, a), IntMap a)
 minViewWithKey (IntMap Empty) = Nothing
-minViewWithKey m = let (k, a) = findMin m
+minViewWithKey m = let (!k, a) = findMin m
                    in Just ((k, a), delete k m)
```

Comment:
That still doesn't do it. The best naive approximation is probably

```haskell
minViewWithKey m
  | Just (k, a) <- lookupMin m
  = Just (k, a, delete k m)
```

but I think ideally we'd pass not `m` but rather the first `Node` of `m` to the deletion function.

Reply:
I must be missing something. Could you explain in more detail what is missing here?

Reply:
That's a peculiar bang pattern that basically says to force `k` *when `a` is forced*. Not likely what you meant.  think you probably meant to force the pair immediately. In each case, the `delete k m` thunk includes the deleted value, which is potentially unfortunate. Separately, I don't like using partial functions like `findMin` internally unless there's a good performance reason to do so.

Reply:
> That's a peculiar bang pattern that basically says to force `k` *when `a` is forced*.

Wow. I completely misunderstood how that worked. `fromAscList` and friends probably ought to be updated, since they also use a nested bang pattern based on a misunderstanding.